contracts/abstract/MasterAwareV2.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "../interfaces/INXMMaster.sol";
+
+abstract contract MasterAwareV2 {
+
+  mapping(uint => address payable) internal internalContracts;
+
+  INXMMaster public master;
+
+  // [todo] Are there any missing contracts here?
+  enum ID {GW, GV, MR, CL, CR, MC, P1, QT, TF, PC, PS, TC, IC, AS}
```

Comment:
should this be centralized at system level, so we don't duplicate it across various components?

Reply:
Theoretically, having this here should be enough. Every contract inheriting this will use the ID enum at it's own deploy time. If more contracts are added it doesn't matter for contracts which are already deployed since it wouldn't use the new contract anyway. This is just used by getInternalContractAddress and the views in the deriving contract that return the callable contract interface. TLDR, it's just syntactic sugar and bytecode optimization as it doesn't inline the same thing in multiple places.

Reply:
The bytecode optimisation is minimal because the memory is not packed. In both cases (enum vs bytes2) there will be a 32-bytes slot used. Bytecode wise, this will benefit of a negligible optimisation in size because we'll have 1 byte instead of 2 bytes in the places where we're using the abbreviation in code (so if we have 5 usages - we save 5 bytes of bytecode).

The only real benefit is a check at compile-time that we don't mispell the abbreviation. This would be caught by integration tests anyway.

On the other hand we would have to upgrade the master contract and add the `getInternalContractAddress(ID id)` function for this to work.

Reply:
I wasn't  talking about the enum at all. The actual bytecode optimisation comes from here:
```  function pool() internal view returns (IPool) {
    return IPool(getInternalContractAddress(ID.P1));
  }
```
Writing `pool()` 2 times instead of `IPool(getInternalContractAddress(ID.P1))` uses a jump to the view instead of inlining the same bytecode twice.